packages/ResizeObserver/src/ResizeObserver.js

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useResizeObserver from "use-resize-observer/polyfilled";
+import debounce from "lodash.debounce";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import * as sc from "./ResizeObserver.styles";
+
+const propTypes = {
+  breakpointLarge: PropTypes.number,
+  breakpointSmall: PropTypes.number,
+  children: PropTypes.node,
+  debounceDelay: PropTypes.number,
+  isFullWidth: PropTypes.bool,
+  isFullHeight: PropTypes.bool,
+  onBreak: PropTypes.func,
+  onResize: PropTypes.func,
+};
+
+const defaultProps = {
+  breakpointLarge: 768,
+  breakpointSmall: 360,
+  children: null,
+  debounceDelay: 200,
+  isFullWidth: true,
+  isFullHeight: false,
+  onBreak: () => {},
+  onResize: () => {},
+};
+
+const MAX_WAIT = 1000;
+
+const ResizeContext = React.createContext();
+
+function useObservedDimensions() {
+  const { width, height } = React.useContext(ResizeContext);
+  return { width, height };
+}
+
+function useBreakpoints() {
+  const { size } = React.useContext(ResizeContext);
+  return { size };
+}
+
+function getSize(width, breakpointSmall, breakpointLarge) {
+  let size = ShirtSizes.MEDIUM;
+  if (breakpointSmall && width <= breakpointSmall) {
```

Comment:
The bounds comparisons seem inconsistent. E.g. for small = 100, large = 200:
```
0 to 100 is small
101 to 199 is medium
200+ is large.
```
I'd expect:
```
0 to 99 small
100 to 199 medium
200+ large
```

Reply:
Yeah, I was thinking that since the `small` breakpoint breaks _to_ small, it should break to the small size at that point.  This is already a bit inconsistent with the way conventional media query breakpoints work, but I thought a `medium` breakpoint would be more ambiguous. But in this case it's probably still reasonable for people to have that mental model and, like you, expect the "break" to the next size up to happen at the breakpoint, so I will adjust it.